ark/phenotyping/cluster.py

```diff
@@ -46,6 +46,58 @@ def winner(sample, weights):
     return winning_coords
 
 
+def batch_winner(samples, weights):
+    """Predict multiple samples at once
+
+    Args:
+        samples (numpy.ndarray):
+            Contains the rows of the pixel matrix to predict
+        weights (numpy.ndarray):
+            A weight matrix of dimensions [num_x, num_y, num_chans]
+
+    Returns:
+        list:
+            The list with indices corresponding to the prediction of the respective sample
+    """
+
+    # only 1 sample provided: expand to 2-D
+    if len(samples.shape) == 1:
+        samples = np.expand_dims(samples, axis=0)
+
+    # collapse the weights for proper subtraction
+    # dimensions will be (m x n) x c
+    # (m x n): number of m and n SOM nodes
+    # c: number of channels
+    weights_collapse = np.reshape(weights, (weights.shape[0] * weights.shape[1], weights.shape[2]))
+
+    # subtract weights from each sample individually, need to reshape back to original
+    # for proper Euclidean distance calculation
+    samples_subtract = samples - weights_collapse[:, None]
+    samples_subtract = np.reshape(samples_subtract, (weights.shape[0], weights.shape[1],
+                                                     samples_subtract.shape[1],
+                                                     samples_subtract.shape[2]))
+
```

Comment:
could be confused about shape, but assuming `samples` has shape `r x c` and weights has shape `m x n x c` I think we can do:
```
samples_subtract = samples.T[np.newaxis, np.newaxis, :, :] - weights[:, :, :, np.newaxis]
```

This should give the same result as above (and if samples is `c x r`, just don't transpose it)

Reply:
The dimensions are slightly off with this method, but I think that's fine. Let me see if I can change the `np.linalg.norm` `axis` parameter, otherwise a call to `swapaxes` shouldn't hurt.